concent_api/core/tests/test_payments_sci_callback.py

```diff
@@ -0,0 +1,192 @@
+        transaction = self._create_unsigned_transaction()
+
+        with mock.patch('core.payments.sci_callback.socket.socket.connect'):
+            with mock.patch('core.payments.sci_callback.send_over_stream'):
+                with mock.patch('core.payments.sci_callback.unescape_stream'):
+                    with mock.patch('middleman_protocol.message.AbstractFrame.deserialize', side_effect=MiddlemanProtocolError):
+                        with self.assertRaises(SCICallbackFrameError):
+                            sci_callback(transaction)
+
+    def test_that_sci_callback_should_raise_exception_on_receiving_incorrectly_signed_golem_message(self):
+        transaction = self._create_unsigned_transaction()
+
+        with mock.patch('core.payments.sci_callback.socket.socket.connect'):
+            with mock.patch('core.payments.sci_callback.send_over_stream'):
+                with mock.patch('core.payments.sci_callback.unescape_stream'):
+                    with mock.patch('middleman_protocol.message.AbstractFrame.deserialize', side_effect=MessageError):
+                        with self.assertRaises(SCICallbackPayloadError):
+                            sci_callback(transaction)
+
+    def test_that_sci_callback_should_raise_exception_when_messages_request_ids_do_not_match(self):
+        transaction = self._create_unsigned_transaction()
+
+        with mock.patch('core.payments.sci_callback.socket.socket.connect'):
+            with mock.patch('core.payments.sci_callback.send_over_stream'):
+                with mock.patch('core.payments.sci_callback.unescape_stream'):
+                    with mock.patch(
+                        'middleman_protocol.message.AbstractFrame.deserialize',
+                        return_value=GolemMessageFrame(payload=self._create_signed_transaction(), request_id=100),
+                    ):
+                        with self.assertRaises(SCICallbackRequestIdError):
+                            sci_callback(transaction)
+
+    def test_that_sci_callback_should_raise_exception_when_response_is_transaction_rejected_or_error_frame(self):
+        transaction = self._create_unsigned_transaction()
+
+        with mock.patch('core.payments.sci_callback.socket.socket.connect'):
+            with mock.patch('core.payments.sci_callback.send_over_stream'):
+                with mock.patch('core.payments.sci_callback.unescape_stream'):
+                    with mock.patch(
+                        'middleman_protocol.message.AbstractFrame.deserialize',
+                        return_value=ErrorFrame(payload=(ErrorCode.UnexpectedMessage, 'error'), request_id=99),
+                    ):
+                        with self.assertRaises(SCICallbackPayloadError):
+                            sci_callback(transaction)
+
+    def test_that_sci_callback_should_raise_exception_when_response_is_not_signed_transaction_golem_message(self):
```

Comment:
hm... "is_message_of_wrong_type"?

Reply:
Actually similar, but mine points out what message type is exactly expected.